server/src/main.ts

```diff
@@ -7,7 +7,7 @@ import { AppModule } from './app.module';
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
   app.enableCors();
-  app.useGlobalPipes(new ValidationPipe());
+  app.useGlobalPipes(new ValidationPipe({ transform: true}));
```

Comment:
tu peux ajouter la transformation implicit

```suggestion
  app.useGlobalPipes(new ValidationPipe({ 
      transform: true,
       transformOptions: {
      enableImplicitConversion: true
    }
      }));
```

et potentiellement une option pour nettoyer les payloads:
- `whitelist` pour retirer silencieusement les propriété invalid
- `forbidNonWhitelisted` pour envoyer une erreur si il y a des champs en trop